### DISCUSSION TITLE
How to get debug information / missing dx features

### DISCUSSION THREAD

**Discussion by GenuineJakob** [KhronosGroup/MoltenVK#1551] [Q&A] [answered]
Hi, I am having trouble getting a game to run with MoltenVK in Crossover/Wine. The developers updated their DirectX requirements and since then the launcher doesn't launch the game anymore and it tells me I have an unsupported DirectX version.
How can I find out which DirectX extensions a game requires?
Is there a log that lists missing extensions in MoltenVK for a running application?

**Comment by cdavis5e** [comment, marked as answer]
By default, MoltenVK dumps the lists of supported features and extensions to `stderr`. If you capture a log from CrossOver ("Run with options..." menu item), you should see lines beginning with `[mvk-info]` in the log file. 

I presume that one of the following is what's happening. Either:
* The game now wants D3D12, but we don't support that yet on Mac; or
* The game wants one of the vendor DLLs that extend D3D11/12, but we don't yet support those on Mac, either.

We'll probably confirm that after you provide logs to CW support. (You did open a ticket or a bug, right?)

**Reply by GenuineJakob** [reply]
Do I need a debug build for this info to appear? Or any logging channels/variables?
If I search for `mvk-info` there is nothing in my log.

I don't want to make this a game-specific question. But for background information, the game is D3D11.
Yes, I opened a ticket, they said it is _a deficiency in MoltenVK_. I asked them which extension is missing, but they could/would not tell. Instead, I got: _there will need to be updates to MoltenVK to enable support for it, which means I wouldn't expect a fix soon, as we do not control that component's development_.

**Reply by cdavis5e** [reply]
MoltenVK has to be built with `-DMVK_LOGGING_ENABLED=1`. This is the default, and I don't see anything in CrossOver's makefiles that changes that.

Logging at the `info` level also has to be enabled. This is also supposed to be the default. You could try setting the environment variable `MVK_CONFIG_LOG_LEVEL=2` and seeing if that helps.

**Reply by cdavis5e** [reply]
There's one other possibility that just occurred to me. The game could want a D3D11 feature that requires one of the Vulkan features that MoltenVK doesn't yet support (see #203).

**Reply by GenuineJakob** [reply]
For some reason, there is nothing about mvk in my log. Do you have any idea why?

[newlog.txt](https://github.com/KhronosGroup/MoltenVK/files/8373001/newlog.txt)

**Reply by billhollings** [reply]
> For some reason, there is nothing about mvk in my log. Do you have any idea why?

Try setting the `MVK_LOGGING_ENABLED=1` environment variable, in addition to `MVK_CONFIG_LOG_LEVEL=2`.

You could also try setting the `MVK_DEBUG=1` environment variable, which also enables logging, and includes shader conversion logging (not that you need that here).

**Reply by GenuineJakob** [reply]
Unfortunately, with
```
Bottle environment variables:
 MVK_LOGGING_ENABLED -> 1
 MVK_CONFIG_LOG_LEVEL -> 2
```
there is still nothing regarding mvk.
[newlog.txt](https://github.com/KhronosGroup/MoltenVK/files/8375157/newlog.txt)

**Reply by cdavis5e** [reply]
According to this log:
> err:vulkan:get_vulkan_driver Wine was built without Vulkan support.

So of course you're not seeing anything from MoltenVK--it's not even loaded. Are you running a 32-bit app? WOW64 Vulkan support is a work in progress.

**Reply by cdavis5e** [reply]
Funny how this came up--as a matter of fact, I was just working on this very thing.

**Reply by GenuineJakob** [reply]
Oh, I see; the launcher is 32bit. The DirectX check is Part of the launcher; MoltenVK is therefore not running during this check. If the check were somehow disabled or bypassed, the game would either start/run fine or crash and give us more information about missing extensions.

Did I understand this correctly?
The game launches fine within Linux, so the 32bit Dx to Vk translation works.

**Reply by cdavis5e** [reply]
> If the check were somehow disabled or bypassed, the game would either start/run fine or crash and give us more information about missing extensions.

Right.

**Reply by GenuineJakob** [reply]
Thank you both for your help. I made a ticket on the developer's forum, asking if we can get a parameter for the launcher to skip the Dx check. How far away are we from 32-bit application support? Are we talking about months or years?

**Reply by cdavis5e** [reply]
Months.
